gnovm/pkg/gnolang/preprocess.go

```diff
@@ -2713,6 +2723,16 @@ func checkType(xt Type, dt Type, autoNative bool) {
 		dt.String()))
 }
 
+// Checks type of NameExpr
+func checkNameExprType(store Store, last BlockNode, nx *NameExpr, t Type) {
```

Comment:
The naming is not very directional, but considering the discussion above, the content of this comment can be ignored.